src/utils/sources-tree/tests/addToTree.spec.js

```diff
@@ -295,9 +296,96 @@ describe("sources-tree", () => {
       expect(formatTree(tree)).toMatchSnapshot();
     });
 
-    it("correctly parses file sources correctly", () => {
+    it("doesnt throw when adding a deeper file", () => {
+      const codeMirror = Map({
+        id: "server1.conn13.child1/37",
+        url: "https://unpkg.com/codemirror@5.1"
+      });
+
+      const xml = Map({
+        id: "server1.conn13.child1/39",
+        url: "https://unpkg.com/codemirror@5.1/mode/xml/xml.js"
+      });
+
+      const tree = createNode("root", "", []);
+
+      addToTree(tree, codeMirror);
+      addToTree(tree, xml);
+
+      expect(formatTree(tree)).toMatchSnapshot();
+    });
+
+    it("gets a source's ancestor directories", function() {
+      const source1 = Map({
+        url: "http://a/b.js",
+        actor: "actor1"
+      });
+
+      const source2 = Map({
+        url: "http://a/c.js",
+        actor: "actor1"
+      });
+
+      const source3 = Map({
+        url: "http://b/c.js",
+        actor: "actor1"
+      });
+
+      const tree = createNode("root", "", []);
+      addToTree(tree, source1);
+      addToTree(tree, source2);
+      addToTree(tree, source3);
+      const paths = getDirectories("http://a/b.js", tree);
+
+      expect(paths[1].path).toBe("/a");
+      expect(paths[0].path).toBe("/a/b.js");
+      expect(formatTree(tree)).toMatchSnapshot();
+    });
+
+    it("handles '?' in target url", function() {
+      const source1 = Map({
+        url: "http://a/b.js",
+        actor: "actor1"
+      });
+
+      const source2 = Map({
+        url: "http://b/b.js",
+        actor: "actor1"
+      });
+
+      const tree = createNode("root", "", []);
+      addToTree(tree, source1);
+      addToTree(tree, source2);
+      const paths = getDirectories("http://a/b.js?key=hi", tree);
```

Comment:
we dont have a getDirectories test at the moment, and it isn't in any of the other source tree test files, so I will leave this here for now